### PR TITLE
[Feature] Admin node refactor

### PR DESCRIPTION
**Pull request by Ghalko**
## Proposal

A cleanup of the admin node interface from improvements accumulated over development.

ADM-82

## Repercussions

More security, using groups and `login_required`. Otherwise no other repercussions.

### CODE REVIEW

File: admin/base/utils.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""
+Utility functions and classes
+"""
+from django.contrib.auth.views import redirect_to_login
+from django.core.exceptions import PermissionDenied
+from django.contrib.auth.mixins import UserPassesTestMixin
+
+
+class OSFAdmin(UserPassesTestMixin):
+    login_url = '/admin/auth/login'
```
Comment (mfraezz) [inline admin/base/utils.py]: Is this necessary? Aside from the lack of a trailing slash, it's the same as `admin_settings.LOGIN_URL`.

Comment (Ghalko) [inline admin/base/utils.py]: You are correct.

File: admin/nodes/urls.py [hunk 2]
```diff
@@ -1,18 +1,18 @@
 from django.conf.urls import url
+from admin.nodes import views
 
-from . import views
 
 urlpatterns = [
     url(r'^$', views.NodeFormView.as_view(),
         name='search'),
-    url(r'^id-(?P<guid>[a-z0-9]+)/$', views.NodeView.as_view(),
+    url(r'^(?P<guid>[a-z0-9]+)/$', views.NodeView.as_view(),
```
Comment (mfraezz) [inline admin/nodes/urls.py]: :+1:

File: admin/nodes/views.py [hunk 3]
```diff
@@ -1,39 +1,29 @@
-from datetime import datetime
+from __future__ import unicode_literals
 
-from django.views.generic import ListView
+from django.views.generic import ListView, DeleteView
+from datetime import datetime
```
Comment (mfraezz) [inline admin/nodes/views.py]: `datetime` is standard library, should be imported in the first section of imports (with but below `__future__`)

File: admin/nodes/views.py [hunk 4]
```diff
@@ -42,16 +32,150 @@ def success_url(self):
...
+            flag = None
+            osf_flag = None
+            message = None
+            if node.is_deleted:
+                node.is_deleted = False
+                node.deleted_date = None
+                flag = NODE_RESTORED
+                message = 'Node {} restored.'.format(node.pk)
+                osf_flag = NodeLog.NODE_CREATED
+            elif not node.is_registration:
```
Comment (mfraezz) [inline admin/nodes/views.py]: So, registrations can be deleted but not undeleted through the admin interface?

File: admin/templates/403.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+{% load static %}
+{% block title %}
+    <title>403</title>
+{% endblock title %}
+{% block content %}
+    <h1><strong>403</strong></h1>
+
+    You do not have permission.
+    <h2>{{ exception | capfirst }}</h2>
+{% endblock content %}
```
Comment (mfraezz) [inline admin/templates/403.html]: `\n` please.

File: admin/nodes/views.py (hunk 4 above)
Comment (Ghalko) [inline admin/nodes/views.py]: No, they can be restored, at the moment, but not deleted.

File: admin/templates/403.html (hunk 5 above)
Comment (Ghalko) [inline admin/templates/403.html]: What does the shorthand mean? For EOF?

Comment (mfraezz) [inline admin/templates/403.html]: A newline at EOF, yes.

screen shot 2016-04-05 at 15 41 35

File: admin/nodes/views.py (hunk 4 above)
Comment (mfraezz) [inline admin/nodes/views.py]: That's what I meant. :+1:

**Comment by mfraezz** [comment]
You have a `flake8` issue causing the build to fail:
>./admin/base/urls.py:1:1: F401 'patterns' imported but unused

Otherwise, looking good. Just want to do some manual testing to make sure the refactored functionality is still good, then it should be RTM :penguin:

File: admin/nodes/views.py (hunk 3 above)
Comment (Ghalko) [inline admin/nodes/views.py]: Done.

File: admin/templates/403.html (hunk 5 above)
Comment (Ghalko) [inline admin/templates/403.html]: Done.

**Comment by mfraezz** [comment]
Thanks for those changes. Manual testing looks good :shipit:

**Pull request merged**